geopandas/geodataframe.py

```diff
@@ -1619,6 +1625,14 @@ def _geodataframe_constructor_sliced(*args, **kwargs):
 
         return _geodataframe_constructor_sliced
 
+    def _constructor_sliced_from_mgr(self, mgr, axes):
+        is_row_proxy = mgr.index.is_(self.columns)
+
+        assert isinstance(mgr, SingleBlockManager)
```

Comment:
Do we need this line for any reason? If we need to do this check, can it be a TypeError instead? Removing it would also allow use removing an import from `pandas.core.internals`.

Reply:
Yeah, I don't think it is needed, this assertion should be guaranteed by pandas (I wrote it in my PR, so I assume that's why Matt copied it)

Reply:
Yep, borrowed from Joris, but also the check below only makes sense if `len(mgr.blocks)==1` i.e. a singleblockmanager. (if the assertion ever not to hold, I don't think we gain a lot by having the assertion vs having some exotic crash in the subsequent pandas code. Happy to remove.